### PR TITLE
Upgrade node version to 12

### LINKED ISSUE #7084: Upgrade Monorepo dependencies to support Node > v10 LTS
The maintenance on v10 ends in April 2021. https://nodejs.org/en/about/releases/.

### PR DESCRIPTION
**Pull request by piersy**
### Description

These changes make the necessary updates to be able to use node version 12 to build and run code in this repo.
Individual commits contain detailed information regarding each change.

One of the changes was to introduce a dependency on @celo/phone-number-privacy-common 1.0.26-dev I think maybe this needs to be published and the dependency adjusted to 1.0.26. It looks like this could be the reason for the google cloud build failing.

I'm also unclear as to what `ci/circleci: protocol-test-release-snapshots Expected — Waiting for status to be reported` is required to make this step complete.

### Tested

yarn install and build work locally & circleci is passing.

### Related issues

- Fixes #7084 

### Backwards compatibility

The backwards compatibility will depend on the specific packages that were updated, I guess some of these changes will be backwards compatible but some wont.

### Documentation

SETUP.md has been updated to instruct people to use node version 12, I think there is some more doc that needs updating though, so I don't consider this part finished yet.

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -56,11 +56,9 @@ Once `nvm` is successfully installed, restart the terminal and run the following
 
 ```bash
 # restart the terminal after installing nvm
-nvm install 8
-nvm install 10
-nvm alias default 10
+nvm install 12
+nvm alias default 12
```
Comment (AlexBHarley) [inline SETUP.md:60]: I use https://github.com/tj/n over nvm and it's great.

**Review by AlexBHarley — COMMENTED** [review]
This looks great! So excited to be on 12, 10 breaks everything else I work on.

File: SETUP.md (hunk 1 above)
Comment (piersy) [reply SETUP.md:60]: Good to know. Are you suggesting we change the doc to reference n?

If so I think that change may deserve it's own ticket. I notice that n needs to be installed via npm which means you need node installed before you can use n, so if we did switch to n in the doc we would need to re-write this doc to cover installing the first version of node manually.

**Comment by piersy** [comment]
@medhak1 I think I need to release a new version of `phone-number-privacy-common` in order to get the google cloud build working, would appreciate your insight here :)

**Comment by piersy** [comment]
Closing, this because this work was covered by #7747 #7833 #7851

**Pull request closed without merging**